pkg/graveler/retention/garbage_collection_manager.go

```diff
@@ -19,9 +19,9 @@ import (
 )
 
 const (
-	configFileSuffixTemplate    = "/%s/retention/gc/rules/config.json"
-	addressesFilePrefixTemplate = "/%s/retention/gc/addresses/"
-	commitsFileSuffixTemplate   = "/%s/retention/gc/commits/run_id=%s/commits.csv"
+	configFileSuffixTemplate    = "%s/retention/gc/rules/config.json"
+	addressesFilePrefixTemplate = "%s/retention/gc/addresses/"
+	commitsFileSuffixTemplate   = "%s/retention/gc/commits/run_id=%s/commits.csv"
```

Comment:
This part fixes another bug we found in the GC-Azure integration! double thanks @Jonathan-Rosenberg :) 
Up until this fix, while trying to get gc rules we got the following error:
```
: invalid namespace
500 Internal Server Error
```
But removing the superfluous "/" fixed it for the following reason - 
Before the fix, "/_lakefs/settings/retention/gc/rules/config.json" was parsed without returning an error to https://github.com/treeverse/lakeFS/blob/a21c0fcfb8d299ac5deb9e4e26bc66e8c8352f35/pkg/block/azure/adapter.go#L91, leading to an incorrect URL resolution trying to resolve it as a fully qualified URI (that includes a namespace). After the fix, "_lakefs/settings/retention/gc/rules/config.json" is resolved as a non-fully qualified URI.